lib/near-contract.js

```diff
@@ -2,19 +2,19 @@ import * as near from "./api";
 export class NearContract {
     deserialize() {
         const rawState = near.storageRead("STATE");
+        let c = this.default();
         if (rawState) {
             const state = JSON.parse(rawState);
-            // reconstruction of the contract class object from plain object
-            let c = this.default();
             Object.assign(this, state);
+            // reconstruction of the contract class object from plain object
             for (const item in c) {
```

Comment:
out of curiosity, why is this `for item in c` and not `for item in state`?

Reply:
if the state is readily initialised, you want to read from it instead of the default, right?

Reply:
@gagdiez We are reading the plain JS object from STATE and converting it to the class JS object because we want to call functions and do everything you expect from this class.

Reply:
but `c` comes from default(), aren't you reading the default state again?

Reply:
@gagdiez No we're not reading the default state. note the following line:
```
                    this[item] = c[item].constructor.deserialize(this[item]);
```
The type of `c` and the data of `this` are used

Reply:
ah, I see, perfect. Thanks for the explanation, I was missing that point.